subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/resolve/service/ScriptPluginUseIntegrationTest.groovy

```diff
@@ -0,0 +1,412 @@
+        !operations.hasOperation("Apply script plugin 'http://localhost:${server.port}/greetings.gradle' to root project 'root'")
+    }
+
+    def "requested script plugin can use classpath dependencies via the buildscript block"() {
+
+        given:
+        file("gradle/other.gradle") << """
+
+            buildscript {
+                repositories { jcenter() }
+                dependencies {
+                    classpath("org.apache.commons:commons-lang3:3.6")
+                }
+            }
+            
+            println(org.apache.commons.lang3.StringUtils.reverse("Gradle"))
+
+        """.stripIndent()
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/other.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help"
+
+        then:
+        output.contains("eldarG")
+
+        and:
+        operations.hasOperation("Apply script plugin 'gradle/other.gradle' to root project 'root'")
+    }
+
+    def "build operations for the application of script plugins requested from files display canonicalized paths relative to build root dir"() {
+
+        given:
+        file("gradle/hello.gradle") << "println('Hello from the other side')"
+
+        and:
+        buildFile << """
+
+            plugins {
+                script("gradle/../gradle/hello.gradle")
+            }
+
+        """.stripIndent()
+
+        when:
+        succeeds "help"
+
+        then:
+        operations.hasOperation("Apply script plugin 'gradle/hello.gradle' to root project 'root'")
+    }
+
+    def "reasonable error message when requested script plugin file does not exist"() {
+
+        given:
+        buildFile << """
+            plugins {
+                script "gradle/other.gradle"
+            }
+        """.stripIndent()
+
+        when:
+        fails "help"
+
+        then:
+        failureCauseContains("Could not read")
+        failureCauseContains("gradle${File.separatorChar}other.gradle' as it does not exist.")
+
+        and:
+        !operations.hasOperation("Apply script plugin 'gradle/other.gradle' to root project 'root'")
+    }
+
+    def "reasonable error message when requested script plugin remote URL does not exist"() {
+
+        given:
+        server.expectGetMissing("/do-not-exist.gradle")
+        server.start()
+
+        and:
+        buildFile << """
+            plugins {
+                script "http://localhost:${server.port}/do-not-exist.gradle"
+            }
+        """.stripIndent()
+
+        when:
+        fails "help"
+
+        then:
+        failureCauseContains("Could not read")
+        failureCauseContains("'http://localhost:${server.port}/do-not-exist.gradle' as it does not exist.")
+
+        and:
+        !operations.hasOperation("Apply script plugin 'http://localhost:${server.port}/do-not-exist.gradle' to root project 'root'")
+    }
+
+
+    def "reasonable error message when the same script plugin request is done twice"() {
```

Comment:
I don't really see the value in this case, given the other duplicate cases above that we ignore. I'd probably just remove this validation.

Reply:
It's consistent with how Gradle behaves with plugin id requests. We should revisit the duplicate request handling strategy in a separate PR.